dashboard/app/models/user.rb

```diff
@@ -1187,12 +1189,27 @@ def visible_script_levels(script)
     end
   end
 
+  def stage_hidden?(stage)
+    section_hidden_stages = SectionHiddenStage.where(
+      stage_id: stage.id,
+      section_id: sections_as_student.pluck(:id)
+    )
+    !section_hidden_stages.empty?
+  end
+
+  def all_stages_hidden?(script)
+    script.stages.detect {|stage| !stage_hidden?(stage)}.nil?
```

Comment:
this seems like a fairly expensive query, particularly for something we're going to be hitting fairly frequently. Is there any way to speed this up?